app/models/git_hub_repository.rb

```diff
@@ -0,0 +1,29 @@
+class GitHubRepository
+  attr_reader :id
+
+  def initialize(id)
+    @id = id
+  end
+
+  def self.create_repository_for_team(org_owner, organization, team_id, repo_name)
+    github_organization = org_owner.github_client.organization(organization.github_id)
+    options             = github_repo_options(github_organization, team_id)
+
+    if (repo = org_owner.github_client.create_repository(repo_name, options))
+      GitHubRepository.new(repo.id)
+    else
+      NullGitHubRepository.new
+    end
+  end
+
+  def self.github_repo_options(github_organization, team_id)
+    {
```

Comment:
@johndbritton how do feel about these being attributes on the `Assignment` model?